src/rapids_singlecell/tools/_umap.py

```diff
@@ -192,8 +204,24 @@ def umap(
     else:
         pre_knn = neighbors["connectivities"]
 
-        if init_pos == "auto":
-            init_pos = "spectral" if n_obs < 1000000 else "random"
+        if isinstance(init_pos, str) and init_pos in adata.obsm:
+            init_coords = adata.obsm[init_pos]
+        elif isinstance(init_pos, str) and init_pos == "paga":
+            init_coords = get_init_pos_from_paga(
+                adata, random_state=random_state, neighbors_key=neighbors_key
+            )
+        elif isinstance(init_pos, str) and init_pos == "auto":
+            init_coords = "spectral" if n_obs < 1000000 else "random"
+        else:
+            init_coords = init_pos
```

Comment:
OK, so neither scanpy’s implementation nor its API is ideal (the scanpy code is older than `match` statements).

I’d generally not advise to make “user-defined string or magic string” APIs as their behavior depends on what the user-supplied data looks like (in this case which keys are in `obsm`).

So the following would be a better implementation, but ideal would be if scanpy and rsc didn’t mix static strings with user-defined strings.

```suggestion
        match init_pos:
            case str() if init_pos in adata.obsm:
                init_coords = adata.obsm[init_pos]
            case "paga":
                init_coords = get_init_pos_from_paga(
                    adata, random_state=random_state, neighbors_key=neighbors_key
                )
            case "auto":
                init_coords = "spectral" if n_obs < 1000000 else "random"
            case _:
                init_coords = init_pos
```